src/embed.ts

```diff
@@ -162,14 +162,14 @@ export abstract class Embed {
      * @param {models.ILoadConfiguration} config
      * @returns {Promise<void>}
      */
-    load(config: models.ILoadConfiguration, embedConfig?: IEmbedConfiguration): Promise<void> {
+    load(config: models.ILoadConfiguration): Promise<void> {
         const errors = models.validateLoad(config);
         if(errors) {
             throw errors;
         }
         
         let loadPath = '/report/load';
-        if(embedConfig && embedConfig.type === 'dashboard') {
+        if(config && (<IEmbedConfiguration>config).type === 'dashboard') {
```

Comment:
Casting here a is sign there is design problem.  It should not be allowed to access the type property.

Since the embed has already been called and the object is constructed it's currently not possible to change types dynamically.  As it is written, I think this would allow a Report object to have dashboard embedded in its iframe which would be very confusing and i'm not sure what problems would occur.

Since embed has already been called at this point in the life-cycle the `type` should be known on the internal config object.  `config: IInternalEmbedConfiguration`.

Try using `this.config.type` instead of the local argument.  Since the type can't be changed within this method it would probably make more sense to have it fixed after the `type` is determined in the `embed` method so there is guarantee the two pieces of information are fixed together.  

I think this problem occurred because we're inside the `Embed` class which has a `load` method intended to be generic for all embeds, but the url path is dependent on the type.  The pieces of information that are specific to each type should be in the concrete classes `Report` or `Dashboard` so they can diverge in their implementation.

Perhaps over ride the `load` method in those classes and each would make the call to the `super.load(....)` using their specific url path.